### DISCUSSION TITLE
onSuccess not triggered Swell payment tokenization

### DISCUSSION THREAD

**Discussion by j2is** [swellstores/community#148] [Help] [answered]
There's a recurring issue where the onSuccess handler is not called in the `swell.payment.tokenize` method. This is the same issue from a little while back: [https://github.com/orgs/swellstores/discussions/23](https://github.com/orgs/swellstores/discussions/23).

Following the advice I was given:
- Settings > Orders > Charge is automatically enabled
- There was no error after the customer completed their bank authorization with 3D Secure 2 
- I coded a delay so the order was submitted after 30s and this didn't capture the payment

**Comment by vstudenichnik-insoft** [comment, marked as answer]
Hi @j2is, could you update swell-js to the latest version and check again?
